mypy/newsemanal/semanal.py

```diff
@@ -1241,24 +1247,14 @@ def get_all_bases_tvars(self,
                 tvars.extend(base_tvars)
         return remove_dups(tvars)
 
-    def setup_class_def_analysis(self, defn: ClassDef) -> None:
+    def prepare_class_def(self, defn: ClassDef, info: Optional[TypeInfo] = None) -> None:
```

Comment:
Documenting this argument would help. (Maybe just explain that some infos are generated by magic defs, so we should use those).